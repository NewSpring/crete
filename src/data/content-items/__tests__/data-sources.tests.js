import ContentItem from '../data-source';

describe('the ContentItem data source', () => {
  it('gets scripture references', async () => {
    const dataSource = new ContentItem();
    dataSource.context = {
      dataSources: {
        Scripture: { getScriptures: jest.fn(() => null) },
        MatrixItem: {
          getItemsFromGuid: () =>
            Promise.resolve([
              {
                attributeValues: {
                  book: { value: 'JOHN' },
                  reference: { value: '10:10' },
                },
              },
              {
                attributeValues: {
                  book: { value: 'JOHN' },
                  reference: { value: '1:1' },
                },
              },
            ]),
        },
      },
    };
    dataSource.request = () => ({
      filter: () => ({
        first: () => Promise.resolve({ value: 'John' }),
      }),
    });
    await dataSource.getContentItemScriptures({
      value: 'FAKEGUID123',
    });
    expect(
      dataSource.context.dataSources.Scripture.getScriptures.mock.calls
    ).toMatchSnapshot();
  });
});
