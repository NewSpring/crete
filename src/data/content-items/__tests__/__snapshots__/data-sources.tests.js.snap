// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ContentItem data source gets scripture references 1`] = `
Array [
  Array [
    "john 1:1, 10:10",
  ],
]
`;
